File: 1827.py
while True:
    try:
        n = int(input())
        ini = n//3
        fim = n - n//3 - 1
        array = []

        for i in range(n):
            array = []
            for j in range(n):
                if i == n//2 and j == n//2:
                    array.append(4)
                elif ini <= i <= fim and ini <= j <= fim:
                    array.append(1)
                elif i + j == n - 1:
                    array.append(3)
                elif i == j:
                    array.append(2)
                else:
                    array.append(0)

            print("".join(str(x) for x in array))
        print("")

    except EOFError:
        break

File: 1827.c
#include <stdio.h>
#include <math.h>

int main() {
    int n, i, j, ini, fim;

    while (scanf("%d", &n) != EOF) {
        ini = n/3;
        fim = n - n/3 - 1;

        for(i = 0; i < n; i++) {
            for(j = 0; j < n; j++) {
                if(i == n/2 && j == n/2)
                    printf("4");
                else if(i >= ini && i <= fim && j >= ini && j <= fim)
                    printf("1");
                else if(i + j == n - 1)
                    printf("3");
                else if(i == j)
                    printf("2");
                else
                    printf("0");
            }

            printf("\n");
        }
        printf("\n");
    }

    return 0;
}
